analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ConditionEvaluatesToConstantBase.cs

```diff
@@ -67,11 +67,16 @@ public override ProgramState ConditionEvaluated(SymbolicContext context)
         return context.State;
     }
 
-    private bool IsNotIgnored(ProgramState state, IOperation operation) =>
-        operation.Kind is not OperationKindEx.Literal
-        && operation.TrackedSymbol(state) is not IFieldSymbol { IsConst: true }
-        && !IsDiscardPattern(operation)
-        && !operation.Syntax.Ancestors().Any(x => IsInsideUsingDeclaration(x) || IsLockStatement(x));
+    private bool IsIgnored(ProgramState state, IOperation operation) =>
+        operation.Kind is OperationKindEx.Literal
+        || IsVarPatternInSwitchArm(operation)
+        || IsDiscardPattern(operation)
+        || operation.TrackedSymbol(state) is IFieldSymbol { IsConst: true }
+        || operation.Syntax.Ancestors().Any(x => IsInsideUsingDeclaration(x) || IsLockStatement(x));
+
+    private static bool IsVarPatternInSwitchArm(IOperation operation) =>
+        operation.AsIsPattern()?.Pattern.WrappedOperation.AsDeclarationPattern()?.MatchesNull is true
+        && operation.Syntax.IsKind(SyntaxKindEx.SwitchExpressionArm);
```

Comment:
This doesn't sound right. What about `if(arg.ToString() is var str)`?

Please add UT

Reply:
Name should get also updated afterwards.

Reply:
We should raise in case of an `if` statement, no? Or am I misunderstanding your question?

Reply:
There is the case of `var` inside a more complex conditional, though. I concluded it is best to be conservative here and not raise on any `var` pattern. I updated the code and added another UT.